assets/js/components/notifications/data-error.js

```diff
@@ -37,32 +35,37 @@ import InvalidCredentialsWarning from './invalid-credentials-warning';
 /**
  * Creates a CTA component when there's a data error. Different wrapper HTML is needed depending on where the CTA gets output, which is determined by the inGrid, fullWidth, and createGrid parameters.
  *
- * @param {string}  moduleName Name of module, translated.
+ * @param {string}  moduleSlug Module slug.
  * @param {string}  error      Description of error.
  * @param {boolean} inGrid     Creates layout to fit within an existing grid with 'cell' classes. Default is half-width grid cells. Default: false.
  * @param {boolean} fullWidth  Creates layout with 'cell--span-12' to be full width. Default: false.
  * @param {boolean} createGrid Adds a full grid layout with padding. Default: false.
- * @param {Object}  errorObj   Error related data.
- *
  * @return {WPElement} CTA component with data error message.
  */
-const getDataErrorComponent = ( moduleName, error, inGrid = false, fullWidth = false, createGrid = false, errorObj = {} ) => {
-	const reconnectUrl = errorObj?.data?.reconnectURL;
-	const description = reconnectUrl ? <ErrorText message={ error } reconnectURL={ reconnectUrl } /> : error;
+function getDataErrorComponent( moduleSlug, error, inGrid = false, fullWidth = false, createGrid = false ) {
+	const modulesData = getModulesData();
+	const module = modulesData[ moduleSlug ];
```

Comment:
This works, but I think we should be able to modify it to rely on `useSelect` and `core/modules` selector `getModule`? Just a minor modernization and one less usage of `getModulesData`. That would also ensure the objects received by `getInsufficientPermissionsErrorDescriptions` are of the same shape for both the legacy and the new code.

Reply:
@felixarntz I am not sure that it will work correctly because we use `getDataErrorComponent` as a regular function. If we used it as a component, we would be able to use hooks there, right?